src/borrower/feed/test/navfeedPV.t.sol

```diff
@@ -0,0 +1,335 @@
+    uint defaultCeilingRatio;
+
+    function setUp() public {
+        // default values
+        defaultThresholdRatio = 8*10**26;                     // 80% threshold
+        defaultCeilingRatio = 6*10**26;                       // 60% ceiling
+        defaultRate = uint(1000000564701133626865910626);     // 5 % day
+
+        feed = new NAVFeedPV();
+        pile = new PileMock();
+        shelf = new ShelfMock();
+        feed.depend("shelf", address(shelf));
+        feed.depend("pile", address(pile));
+        
+        feed.file("riskGroup",
+            0,                                      // riskGroup:       0
+            8*10**26,                               // thresholdRatio   70%
+            6*10**26,                               // ceilingRatio     60%
+            uint(1000000564701133626865910626)      // interestRate     5% per year
+        );
+
+         feed.file("riskGroup",
+            1,                                      // riskGroup:       0
+            8*10**26,                               // thresholdRatio   80%
+            7*10**26,                               // ceilingRatio     70%
+            ONE                                     // interestRate     1.0
+        );
+        feed.file("riskGroup",
+            2,                                      // riskGroup:       0
+            8*10**26,                               // thresholdRatio   90%
+            8*10**26,                               // ceilingRatio     80%
+            uint(1000000003593629043335673583)      // interestRate     12% per year
+        );
+    }
+
+    function testNFTValues() public {
+        bytes32 nftID = feed.nftID(address(1), 1);
+        uint value = 100 ether;
+        feed.update(nftID, value);
+
+        uint loan = 1;
+        shelf.setReturn("shelf", address(1), 1);
+
+        assertEq(feed.nftValues(nftID), 100 ether);
+        assertEq(feed.threshold(loan), 80 ether);
+        assertEq(feed.ceiling(loan), 60 ether);
+    }
+
+    function testCreditLineCeilingOutstandingDebt() public {
+        bytes32 nftID = feed.nftID(address(1), 1);
+        uint riskGroup = 1;
+        uint value = 100 ether;
+        uint loan = 1;
+        uint debt = 40 ether;
+        uint maxCeiling = 70 ether;
+        feed.update(nftID, value, riskGroup);
+
+        shelf.setReturn("shelf", address(1), 1);
+        assertEq(feed.ceiling(loan), maxCeiling);
+        pile.incDebt(loan, debt);
+        assertEq(feed.ceiling(loan), safeSub(maxCeiling, debt));
+    }
+
+    function testCreditLineCeilingOutstandingDebtBiggerCeiling() public {
+        bytes32 nftID = feed.nftID(address(1), 1);
+        uint riskGroup = 1;
+        uint loan = 1;
+        uint value = 100 ether;
+        uint debt = 71 ether;
+        uint maxCeiling = 70 ether;
+        feed.update(nftID, value, riskGroup);
+
+        shelf.setReturn("shelf", address(1), 1);
+        assertEq(feed.ceiling(loan), maxCeiling);
+        pile.incDebt(loan, debt);
```

Comment:
Okay, to summarize. The actual `ceiling` check happens in the borrow in the nav feed which is triggered by the `borrower` subscriber event in the shelf.

We need to ensure that the subscriber is always set on the shelf.

Reply:
yes, the checks happens in the NAV. If the address wouldn't be set, the check would revert anyway, right